enhancedautoclicker.js

```diff
@@ -267,16 +269,27 @@ function autoGym() {
             }*/
 
             if (App.game.gameState != GameConstants.GameState.gym) {
-                //If the content is a Gym or league champion and we unlocked it we fight
-                if ((player.town().content[gymSelect] instanceof Gym && player.town().content[gymSelect].isUnlocked()) || (player.town().content[gymSelect] instanceof Champion && player.town().content[gymSelect].isUnlocked())){
-                    GymRunner.startGym(player.town().content[gymSelect])
-                }
-                else {
-                    //Otherwise we try to fight the previous gyms (elite 4)
-                    for (var i = player.town().content.length - 1; i >= 0; i--){
-                        if ((player.town().content[i] instanceof Gym && player.town().content[i].isUnlocked()) || (player.town().content[i] instanceof Champion && player.town().content[i].isUnlocked())){
-                            GymRunner.startGym(player.town().content[i])
-                            break
+                //If "All" is selected, then go through list of league from 0-4
+                if(gymSelect === 5) {
+                    var allSelectedGym = localStorage.getItem('allSelectedGym')
+                    GymRunner.startGym(player.town().content[allSelectedGym])
+                    allSelectedGym++
+                    if(allSelectedGym === 5) {
+                        allSelectedGym = 0
+                    }
+                    localStorage.setItem("allSelectedGym", allSelectedGym);
```

Comment:
localStorage should only be used for variables you need to remember between sessions but this one is reset on startup, consider using a local variable instead. Reason why is localStorage has a limit of variables you can save locally so it should be used sparsely.

Reply:
Updated to use variable. Haven't tested it